R/client.R

```diff
@@ -48,3 +62,51 @@ get_service_name <- function(service_url) {
     EMODnetWFS::emodnet_wfs$service_name[EMODnetWFS::emodnet_wfs$service_url == service_url]
 }
 
+
+# Checks if there is internet and performs an HTTP GET request
+perform_http_request <- function(service_url){
+    usethis::ui_oops("WFS client creation failed.")
+    usethis::ui_info("Service: {usethis::ui_value(service_url)}")
+
+    if(!curl::has_internet()){
+        usethis::ui_info("Reason: There is no internet connection")
+        return(NULL)
+    }
+
+    service_url %>%
+        paste0("?request=GetCapabilities") %>%
+        httr::GET()
+
+}
+
+
+# Checks if there is internet connection and HTTP status of the service
+# If fails returns FALSE, if there is internet and HTTP status is successful then return TRUE
+check_service <- function(request){
+
+    # Raise the GetCapabilities throws a bad HTTP status
+    if(httr::http_error(request)){
+        usethis::ui_info("HTTP Status: {crayon::red(httr::http_status(request)$message)}")
+        usethis::ui_line()
+
+        # Check if monitor tool is up, and then ask if wants to browse the app
+        if(interactive() & !is.null(curl::nslookup("monitor.emodnet.eu", error = FALSE))){
```

Comment:
Maybe here you could use an explanatory variable just for readability (not sure if useful!)

```r
is_monitor_up <- !is.null(curl::nslookup("monitor.emodnet.eu", error = FALSE))
if (interactive() && is_monitor:up() {
```

Reply:
Really like @maelle's suggestion here +1

Reply:
https://twitter.com/JennyBryan/status/1412140590842597385 :-)

Reply:
Of course! 😜

Reply:
Absolutely! Will do. Thanks!